jolt-core/src/jolt/instruction/mulh.rs

```diff
@@ -0,0 +1,250 @@
+use tracer::{ELFInstruction, RVTraceRow, RegisterState, RV32IM};
+
+use super::VirtualInstructionSequence;
+use crate::jolt::instruction::{
+    add::ADDInstruction, movsign::MOVSIGNInstruction, mulhu::MULHUInstruction,
+    mulu::MULUInstruction, JoltInstruction,
+};
+
+pub struct MULHInstruction<const WORD_SIZE: usize>;
+
+impl<const WORD_SIZE: usize> VirtualInstructionSequence for MULHInstruction<WORD_SIZE> {
+    fn virtual_sequence(trace_row: RVTraceRow) -> Vec<RVTraceRow> {
+        assert_eq!(trace_row.instruction.opcode, RV32IM::MULH);
+        // MULH operands
+        let x = trace_row.register_state.rs1_val.unwrap();
+        let y = trace_row.register_state.rs2_val.unwrap();
+        // MULH source registers
+        let r_x = trace_row.instruction.rs1;
+        let r_y = trace_row.instruction.rs2;
+        // Virtual registers used in sequence
+        let v_sx = Some(32);
```

Comment:
Let's split out constants::VIRTUAL_REGISTER_COUNT from constants::REGISTER_COUNT and add a constants::virtual_register_index(index: usize) -> usize { REGISTER_COUNT + index};